[Bug]: Error: Insufficient number of substitutions for key "insufficientCurrencyBuyOrReceive" with locale "en"

### Describe the bug


https://user-images.githubusercontent.com/53189696/206752920-b70c8fd3-c905-49fa-9bc4-387af52210ff.mov



### Steps to reproduce

1. Login extension
2. Switch to localhost
3. Click the Send button on the overview page
4. Enter an amount thats more than the native currency held on Locahost and Ethereum Mainnet
5. Switch to Ethereum Mainnet


### Error messages or log output

```shell
sentry-install.js:76 Error: Insufficient number of substitutions for key "insufficientCurrencyBuyOrReceive" with locale "en"
  at i18n-helper.js:80:23
  at Array.map (<anonymous>)
  at n.getMessage (i18n-helper.js:65:36)
  at i18n.js:20:7
  at x (gas-display.js:470:22)
  at ha (react-dom.production.min.js:154:1)
  at Qa (react-dom.production.min.js:174:1)
  at Cu (react-dom.production.min.js:261:1)
  at El (react-dom.production.min.js:230:1)
  at bl (react-dom.production.min.js:229:1)
  at fl (react-dom.production.min.js:223:1)
  at react-dom.production.min.js:121:1
  at n.unstable_runWithPriority (scheduler.production.min.js:18:1)
  at Ko (react-dom.production.min.js:120:1)
  at Zo (react-dom.production.min.js:121:1)
  at Jo (react-dom.production.min.js:120:1)
  at pl (react-dom.production.min.js:224:1)
  at Object.notify (Subscription.js:25:1)
  at t.notifyNestedSubs (Subscription.js:98:1)
  at t.handleChangeWrapper (Subscription.js:103:1)
  at y (redux.js:304:1)
  at index.js:14:1
  at dispatch (redux.js:667:1)
  at actions.js:2007:5
```


### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

> ### Describe the bug
> 
> 
> https://user-images.githubusercontent.com/53189696/206752920-b70c8fd3-c905-49fa-9bc4-387af52210ff.mov
> 
> 
> 
> ### Steps to reproduce
> 
> 1. Login extension
> 2. Switch to localhost
> 3. Click the Send button on the overview page
> 4. Enter an amount thats more than the native currency held on Locahost and Ethereum Mainnet
> 5. Switch to Ethereum Mainnet
> 
> 
> ### Error messages or log output
> 
> ```shell
> sentry-install.js:76 Error: Insufficient number of substitutions for key "insufficientCurrencyBuyOrReceive" with locale "en"
>   at i18n-helper.js:80:23
>   at Array.map (<anonymous>)
>   at n.getMessage (i18n-helper.js:65:36)
>   at i18n.js:20:7
>   at x (gas-display.js:470:22)
>   at ha (react-dom.production.min.js:154:1)
>   at Qa (react-dom.production.min.js:174:1)
>   at Cu (react-dom.production.min.js:261:1)
>   at El (react-dom.production.min.js:230:1)
>   at bl (react-dom.production.min.js:229:1)
>   at fl (react-dom.production.min.js:223:1)
>   at react-dom.production.min.js:121:1
>   at n.unstable_runWithPriority (scheduler.production.min.js:18:1)
>   at Ko (react-dom.production.min.js:120:1)
>   at Zo (react-dom.production.min.js:121:1)
>   at Jo (react-dom.production.min.js:120:1)
>   at pl (react-dom.production.min.js:224:1)
>   at Object.notify (Subscription.js:25:1)
>   at t.notifyNestedSubs (Subscription.js:98:1)
>   at t.handleChangeWrapper (Subscription.js:103:1)
>   at y (redux.js:304:1)
>   at index.js:14:1
>   at dispatch (redux.js:667:1)
>   at actions.js:2007:5
> ```
> 
> 
> ### Version
> 
> 10.24.0
> 
> ### Build type
> 
> None
> 
> ### Browser
> 
> Chrome
> 
> ### Operating system
> 
> MacOS
> 
> ### Hardware wallet
> 
> _No response_
> 
> ### Additional context
> 
> _No response_

We'd be happy to look into this for you. kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

--- a/ui/pages/send/gas-display/gas-display.js
+++ b/ui/pages/send/gas-display/gas-display.js
@@ -66,7 +66,7 @@ export default function GasDisplay({ gasError }) {
   const isInsufficientTokenError =
     draftTransaction?.amount.error === INSUFFICIENT_TOKENS_ERROR;
   const editingTransaction = unapprovedTxs[draftTransaction.id];
-  const currentNetworkName = networkName ? networkName : currentProvider.nickname;
+  const currentNetworkName = networkName || currentProvider.nickname;
 
   const transactionData = {
     txParams: {

--- a/ui/pages/send/gas-display/gas-display.js
+++ b/ui/pages/send/gas-display/gas-display.js
@@ -66,6 +66,7 @@ export default function GasDisplay({ gasError }) {
   const isInsufficientTokenError =
     draftTransaction?.amount.error === INSUFFICIENT_TOKENS_ERROR;
   const editingTransaction = unapprovedTxs[draftTransaction.id];
+  const currentNetworkName = networkName ? networkName : currentProvider.nickname;
 
   const transactionData = {
     txParams: {
@@ -304,7 +305,7 @@ export default function GasDisplay({ gasError }) {
           ]}
         />
       </Box>
-      {(gasError || isInsufficientTokenError) && (
+      {(gasError || isInsufficientTokenError) && currentNetworkName && (
         <Box
           className="gas-display__warning-message"
           data-testid="gas-warning-message"
@@ -322,7 +323,7 @@ export default function GasDisplay({ gasError }) {
                   <Typography variant={TYPOGRAPHY.H7} align="left">
                     {t('insufficientCurrencyBuyOrReceive', [
                       nativeCurrency,
-                      networkName ?? currentProvider.nickname,
+                      currentNetworkName,
                       <Button
                         type="inline"
                         className="confirm-page-container-content__link"
@@ -349,7 +350,7 @@ export default function GasDisplay({ gasError }) {
                   <Typography variant={TYPOGRAPHY.H7} align="left">
                     {t('insufficientCurrencyBuyOrReceive', [
                       draftTransaction.asset.details?.symbol ?? nativeCurrency,
-                      networkName ?? currentProvider.nickname,
+                      currentNetworkName,
                       `${t('buyAsset', [
                         draftTransaction.asset.details?.symbol ??
                           nativeCurrency,

